src/file/source/file.rs

```diff
@@ -114,8 +116,15 @@ where
             .and_then(|base| pathdiff::diff_paths(&filename, base))
             .unwrap_or_else(|| filename.clone());
 
-        // Read contents from file
-        let text = fs::read_to_string(filename)?;
+        let mut text = String::new();
+
+        DecodeReaderBytesBuilder::new()
+            // On Windows, we need to check for added Byte Order Mark (BOM). To
+            // do this, we don't specify an encoding, and enable BOM sniffing.
+            .encoding(None)
+            .bom_sniffing(true)
+            .build(std::fs::File::open(filename)?)
+            .read_to_string(&mut text)?;
```

Comment:
if I'm reading the docs correctly, this does more than BOM sniffing / stripping but also handles other encodings.  That should be a separate question, discussed in an issue first.  This also then feels like a heavy weight dependency just for handling BOM.